test/Sentry.Tests/Protocol/ScopeExtensionsTests.cs

```diff
@@ -547,6 +573,57 @@ public void AddAttachment_FromFile_ArgumentsResolvedCorrectly()
             Assert.Equal(12, attachment?.Length);
         }
 
+        [Fact]
+        public async Task AddAttachment_FromFile_RespectsLimits()
+        {
+            // Arrange
+            var logger = new AccumulativeDiagnosticLogger();
+            _fixture.ScopeOptions.DiagnosticLogger = logger;
+            _fixture.ScopeOptions.Debug = true;
+            _fixture.ScopeOptions.MaxAttachmentSize = 1000; // 1kb
+
+            using var tempDir = new TempDirectory();
+            var filePath = Path.Combine(tempDir.Path, "MyFile.bin");
+
+            using (var fileStream = File.Create(filePath))
+            {
+                var random = new Random();
+                await random.WriteToStreamAsync(fileStream, 1005);
+            }
+
+            var scope = _fixture.GetSut();
+
+            // Act
+            scope.AddAttachment(filePath);
+
+            // Assert
+            Assert.Empty(scope.Attachments);
+            Assert.Contains(logger.Entries, e =>
+                e.Message == "Attachment '{0}' dropped because it's too large." &&
+                e.Args[0].ToString() == "MyFile.bin"
+            );
+        }
+
+        [Fact]
+        public void AddAttachment_FromFile_FileDoesNotExist_Drops()
```

Comment:
Do we have a file that throws to read?

Reply:
That will just throw during envelope serialization.

Reply:
I meant: Do we have a test that covers that caase? Ia the whole envelope thrown away or can we skip just the item like in Java and Objective-C?

Reply:
We have this test in `BackgroundWorkerTests`:

![image](https://user-images.githubusercontent.com/1935960/104212170-984f3c80-543d-11eb-99c9-939e0410d045.png)

This verifies that an exception thrown when sending envelope is swallowed. It would go via `Envelope.SerializeAsync` -> `HttpTransport.SendEnvelopeAsync` -> `BackgroundWorker.EnqueueEnvelope`.